upload/common/uploaded_file.py

```diff
@@ -29,39 +30,68 @@ def from_s3_key(cls, upload_area, s3_key):
         return cls(upload_area, s3object=s3object)
 
     def __init__(self, upload_area, name=None, content_type=None, data=None, s3object=None):
+        # Properties persisted in DB
+        self._properties = {
+            "id": None,
+            "s3_key": None,
+            "s3_etag": None,
+            "upload_area_id": upload_area.db_id,
+            "name": None,
+            "size": None
+        }
         self.upload_area = upload_area
-        self.s3obj = None
-        self.name = name
         self.checksums = {}
-        self.db = UploadDB()
+        # internals
+        self._s3obj = None
+        # utility:
+        self._db = UploadDB()
+
         if name and data and content_type:
-            self._create_s3_object(data, content_type)
-            self._fetch_or_create_db_record()
+            self._s3_create(name, data, content_type)
+            self._populate_properties_from_s3_object()
+            self._db_create()
```

Comment:
do we need to call self._db_load() after this

Reply:
`_db_create()` syncs the DB with the state of this object, *and* captures the id and inserts it back in `_properties`.  This is moot however as all this got refactored in a later commit.